File: backend/app/schemas/watchlist.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class WatchlistCreate(BaseModel):
    symbol: str

class WatchlistItem(BaseModel):
    id: int
    symbol: str
    created_at: datetime
    
    class Config:
        from_attributes = True

class WatchlistResponse(BaseModel):
    id: int
    symbol: str
    name: str
    current_price: float
    previous_close: float
    change: float
    change_percent: float
    market_cap: str
    created_at: datetime
    
    class Config:
        from_attributes = True

File: frontend/src/lib/types/watchlist.ts
// defines the structure of a stock in the watchlist
export interface WatchlistStock {
    id: number;
    symbol: string;
    name: string;
    current_price: number;
    previous_close: number;
    change: number;
    change_percent: number;
    market_cap: string;
    created_at: string;
}
